### PR TITLE
chore: remove usage stats

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/10259

### AC

- [ ] Usage stats has been safely removed

### CODE REVIEW

File: packages/client/components/TeamsLimitExceededNotification.tsx [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-
-  const onActionClick = () => {
-    history.push(`/usage`)
-  }
-
-  return (
-    <NotificationTemplate
-      avatar={orgPicture || defaultOrgAvatar}
-      message={`"${orgName}" is over the limit of ${Threshold.MAX_STARTER_TIER_TEAMS} free teams. Action is needed.`}
-      action={<NotificationAction label={'See Usage'} onClick={onActionClick} />}
```
Comment (nickoferrall) [inline packages/client/components/TeamsLimitExceededNotification.tsx:39]: Teams limit directs users to the Usage page, so I think it's safe to remove

**Comment by nickoferrall** [comment]
Hey @mattkrick, just bumping the review request on this one 🙏

**Comment by nickoferrall** [comment]
Hey @mattkrick , bumping this one as it's getting stale

**Review by mattkrick — APPROVED** [review]
looks great!

**Pull request merged**